tools/xtask/src/tasks/license.rs

```diff
@@ -0,0 +1,80 @@
+use std::fs;
+use std::path::{Path, PathBuf};
+
+use anyhow::{anyhow, Result};
+use clap::Parser;
+
+use tracing::info;
+
+use cargo_metadata::Metadata;
+
+#[derive(Parser)]
+pub struct LicenseArgs {}
+
+pub fn run_license(_args: LicenseArgs, workspace: Metadata) -> Result<()> {
+    const LICENSE_FILES: &[&str] = &["LICENSE-MIT"];
+    let default_license = workspace.workspace_root.join(LICENSE_FILES[0]);
+
+    for package in workspace.workspace_packages() {
+        let crate_path = package
+            .manifest_path
+            .parent()
+            .ok_or_else(|| anyhow!("no crate directory for '{}'", package.name))?;
+
+        if let Some((symlink_path, license_index)) =
+            get_first_license_symlink_path(crate_path, LICENSE_FILES)
+        {
+            let root_license_path = pathdiff::diff_paths(
+                workspace.workspace_root.join(LICENSE_FILES[license_index]),
+                crate_path,
+            )
+            .expect("failed to create relative path for root license");
+            if symlink_path.is_symlink() {
+                let target = fs::read_link(&symlink_path)?;
+                if target != root_license_path {
+                    info!("updating symlink for '{}'", package.name);
+                    fs::remove_file(&symlink_path)?;
+                    create_symlink(&root_license_path, &symlink_path)?;
+                }
+            } else {
+                info!("replacing file with symlink for '{}'", package.name);
+                fs::remove_file(&symlink_path)?;
+                create_symlink(&root_license_path, &symlink_path)?;
+            }
+        } else {
+            info!("creating license symlink for '{}'", package.name);
```

Comment:
I think the handling of symbolic link updates should be moved to a separate `handle_symlink_update`